spec/unit/http/client_spec.rb

```diff
@@ -45,6 +48,38 @@
         client.connect(uri)
       }.to raise_error(Puppet::HTTP::ConnectionError, %r{^Request to https://www.example.com timed out connect operation after .* seconds})
     end
+
+    it 'connects using the default ssl context' do
+      expect(pool).to receive(:with_connection) do |_, verifier|
+        expect(verifier.ssl_context).to equal(puppet_context)
+      end
+
+      client.connect(uri)
+    end
+
+    it 'connects using a specified ssl context' do
+      other_context = Puppet::SSL::SSLContext.new
+
+      expect(pool).to receive(:with_connection) do |_, verifier|
+        expect(verifier.ssl_context).to equal(other_context)
+      end
+
+      client.connect(uri, ssl_context: other_context)
+    end
+
+    it 'connects using the system store' do
+      expect(pool).to receive(:with_connection) do |_, verifier|
+        expect(verifier.ssl_context).to eq(system_context)
```

Comment:
I think you want `equal(system_context)` here

Reply:
Updated